### PR TITLE
BUG: create new MI from MultiIndex._get_level_values

### LINKED ISSUE #33131: BUG: Copying PeriodIndex levels on MultiIndex loses weakrefs
As per [comment](https://github.com/pandas-dev/pandas/pull/32669#issuecomment-605525216) by @jacobaustin123:

```python
import pandas as pd
idx = pd.MultiIndex.from_arrays([pd.PeriodIndex([pd.Period("2019Q1"), pd.Period("2019Q2")], name='b')])
idx2 = pd.MultiIndex.from_arrays([idx._get_level_values(level) for level in range(idx.nlevels)])
all(x.is_monotonic for x in idx2.levels) # raises an error
```

#### Problem description

The weakly referenced PeriodIndex er dropped before intended, so the ``PeriodEngine`` gets a ``None`` instead of the PeriodIndex.

#### Expected Output

The above should return True.

### PR DESCRIPTION
**Pull request by topper-123**
- [x] closes #33131
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Closes #33131, a weakref was released too early.

### CODE REVIEW

File: pandas/tests/indexes/multi/test_get_level_values.py [hunk 1]
```diff
@@ -89,3 +89,17 @@ def test_get_level_values_na():
...
+    # This test can probably be removed when PeriodIndex._engine is removed.
+    from pandas import Period, PeriodIndex
+
+    idx = MultiIndex.from_arrays(
+        [PeriodIndex([Period("2019Q1"), Period("2019Q2")], name="b")]
+    )
+    idx2 = MultiIndex.from_arrays(
+        [idx._get_level_values(level) for level in range(idx.nlevels)]
+    )
+    assert all(x.is_monotonic for x in idx2.levels) is True
```
Comment (MomIsBestFriend) [inline pandas/tests/indexes/multi/test_get_level_values.py]: Question not a blocker.

Is the ```is True``` necessary?

Comment (topper-123) [reply pandas/tests/indexes/multi/test_get_level_values.py]: Yeah, in this case it's not needed. I'm in the habit of testing properties with a strict ``is`` test, where possible. I'll remove it, but will wait to see if other comments need addressing.

**Review by jreback — APPROVED** [review]
lgtm. if you'd add a whatsnew note. merge on green.

**Comment by jbrockmendel** [comment]
If PeriodEngine.get_indexer is ever called it will raise AttributeError, ditto get_pad_indexer, get_backfill indexer.

**Comment by topper-123** [comment]
@jreback , this is a regression since v1.0, causes by #32669, so a whatsnew isn't needed here.

@jbrockmendel , it looks like it also did that in v1.0?:

```python
>>> idx = pd.MultiIndex.from_arrays([pd.PeriodIndex([pd.Period("2019Q1"), pd.Period("2019Q2")], name='b')])
>>> idx2 = pd.MultiIndex.from_arrays([idx._get_level_values(level) for level in range(idx.nlevels)])
>>> idx2.levels[0]._engine.get_indexer(idx.levels[0])
AttributeError: 'super' object has no attribute '_ensure_mapping_populated'  # both 1.0 and master
```

**Comment by jreback** [comment]
thanks @topper-123

**Pull request merged**